File: python/meteroid/models/invoice_line_item.py
# this file is @generated
import dataclasses
import typing as t
from decimal import Decimal

from ..serialization import BaseModel
from .sub_line_item import SubLineItem


@dataclasses.dataclass
class InvoiceLineItem(BaseModel):
    amount_total: int

    end_date: str

    name: str

    start_date: str

    sub_line_items: t.List[SubLineItem]

    tax_rate: Decimal

    description: t.Optional[str] = None

    quantity: t.Optional[Decimal] = None

    unit_price: t.Optional[Decimal] = None

File: go/invoice_line_item.go
// this file is @generated
package meteroid

type InvoiceLineItem struct {
	AmountTotal int64 `json:"amount_total"`

	Description *string `json:"description,omitempty"`

	EndDate string `json:"end_date"`

	Name string `json:"name"`

	Quantity *string `json:"quantity,omitempty"`

	StartDate string `json:"start_date"`

	SubLineItems RequiredSlice[SubLineItem] `json:"sub_line_items"`

	TaxRate string `json:"tax_rate"`

	UnitPrice *string `json:"unit_price,omitempty"`
}
